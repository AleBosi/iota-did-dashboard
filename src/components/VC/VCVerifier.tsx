import React from "react";

interface VCData {
  subject: string;
  type: string;
  value: string;
  proof?: {
    hash: string;
    signature: string;
  };
}

interface Props {
  vc: VCData;
  verify: (vc: VCData) => boolean;
}

export default function VCVerifier({ vc, verify }: Props) {
  const isValid = verify(vc);

  return (
    <div className="flex items-center gap-2">
      <span>
        Stato firma:{" "}
        {isValid ? (
          <span className="text-green-600 font-bold">✅ Valida</span>
        ) : (
          <span className="text-red-600 font-bold">❌ Non valida</span>
        )}
      </span>
    </div>
  );
}
