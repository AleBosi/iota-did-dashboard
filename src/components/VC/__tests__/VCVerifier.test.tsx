import { render, screen } from "@testing-library/react";
import VCVerifier from "../VCVerifier";

const mockVC = {
  subject: "did:iota:123",
  type: "Compliance",
  value: "Certificato OK",
  proof: {
    hash: "hash123",
    signature: "sig123"
  }
};

describe("VCVerifier component", () => {
  it("mostra ✅ Valida in verde quando verify ritorna true", () => {
    const verifyMock = jest.fn().mockReturnValue(true);
    render(<VCVerifier vc={mockVC} verify={verifyMock} />);
    expect(screen.getByText(/Valida/)).toHaveClass("text-green-600");
    expect(screen.getByText(/✅/)).toBeInTheDocument();
    expect(screen.queryByText(/Non valida/)).toBeNull();
    expect(verifyMock).toHaveBeenCalledWith(mockVC);
  });

  it("mostra ❌ Non valida in rosso quando verify ritorna false", () => {
    const verifyMock = jest.fn().mockReturnValue(false);
    render(<VCVerifier vc={mockVC} verify={verifyMock} />);
    expect(screen.getByText(/Non valida/)).toHaveClass("text-red-600");
    expect(screen.getByText(/❌/)).toBeInTheDocument();
    expect(screen.queryByText(/Valida/)).toBeNull();
    expect(verifyMock).toHaveBeenCalledWith(mockVC);
  });

  it("matcha lo snapshot", () => {
    const verifyMock = jest.fn().mockReturnValue(true);
    const { asFragment } = render(<VCVerifier vc={mockVC} verify={verifyMock} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
