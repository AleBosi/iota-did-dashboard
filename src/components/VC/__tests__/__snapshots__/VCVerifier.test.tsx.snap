// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`VCVerifier component matcha lo snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex items-center gap-2"
  >
    <span>
      Stato firma: 
      <span
        class="text-green-600 font-bold"
      >
        âœ… Valida
      </span>
    </span>
  </div>
</DocumentFragment>
`;
